src/main/java/core/basesyntax/servicehandler/PurchaseHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.servicehandler;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseHandler implements FruitOperationHandler {
+    private final Map<String, Integer> fruitStorage;
+
+    public PurchaseHandler(Map<String, Integer> fruitStorage) {
+        this.fruitStorage = fruitStorage;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        fruitStorage.put(transaction.getFruit(),
+                fruitStorage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
Consider adding validation to ensure that the resulting quantity after a purchase does not become negative. This will help prevent logical errors and ensure the integrity of the fruit storage.